src/beliefpropagation.jl

```diff
@@ -47,53 +72,56 @@ DO a single update of a message tensor using the current subgraph and the incomi
 function update_mt(
   tn::ITensorNetwork,
   subgraph_vertices::Vector,
-  mts::Vector{ITensor};
-  contraction_sequence::Function=tn -> contraction_sequence(tn; alg="optimal"),
+  mts::Vector{ITensorNetwork};
+  contract_kwargs=(;),
 )
-  contract_list = [mts; [tn[v] for v in subgraph_vertices]]
+  contract_list = ITensorNetwork[mts; ITensorNetwork([tn[v] for v in subgraph_vertices])]
 
-  new_mt = if isone(length(contract_list))
+  tn = if isone(length(contract_list))
     copy(only(contract_list))
   else
-    contract(contract_list; sequence=contraction_sequence(contract_list))
+    reduce(⊗, contract_list)
+  end
+
+  contract_output = contract(tn; contract_kwargs...)
+  itn = if typeof(contract_output) == ITensor
+    ITensorNetwork(contract_output)
+  else
+    first(contract_output)
   end
-  return normalize!(new_mt)
+  normalize!.(vertex_data(itn))
+
+  return itn
 end
 
 function update_mt(
-  tn::ITensorNetwork, subgraph::ITensorNetwork, mts::Vector{ITensor}; kwargs...
+  tn::ITensorNetwork, subgraph::ITensorNetwork, mts::Vector{ITensorNetwork}; kwargs...
 )
   return update_mt(tn, vertices(subgraph), mts; kwargs...)
 end
 
 """
 Do an update of all message tensors for a given ITensornetwork and its partition into sub graphs
 """
-function update_all_mts(
-  tn::ITensorNetwork,
-  mts::DataGraph;
-  contraction_sequence::Function=tn -> contraction_sequence(tn; alg="optimal"),
-)
+function update_all_mts(tn::ITensorNetwork, mts::DataGraph; contract_kwargs=(;))
```

Comment:
Can we just call this `belief_propagation_step` or `belief_propagation_iteration` instead of `update_all_mts`? I guess this is a single step of BP? And can we call the outer loop just `belief_propagation`?